src/utilities/printf.c

```diff
@@ -128,23 +128,32 @@ hypre_ndigits( HYPRE_BigInt number )
 /* printf functions */
 
 HYPRE_Int
-hypre_printf( const char *format, ...)
+hypre_vprintf( const char *format, va_list ap )
 {
-   va_list   ap;
-   char     *newformat;
    HYPRE_Int ierr = 0;
+   char *newformat;
 
-   va_start(ap, format);
    new_format(format, &newformat);
    ierr = vprintf(newformat, ap);
    free_format(newformat);
-   va_end(ap);
-
    fflush(stdout);
 
    return ierr;
 }
 
+HYPRE_Int
+hypre_printf( const char *format, ...)
+{
+   va_list   ap;
+   HYPRE_Int ierr = 0;
+
+   va_start(ap, format);
+   ierr = hypre_vprintf(format, ap);
```

Comment:
Call `vprintf` directly here.  I don't think there is any meaningful reason to write a `hypre_vprintf` function.  We won't use it anywhere else in the code, and it really only saves a few lines of code anyway.